adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/impl/VertxBasedMqttProtocolAdapter.java

```diff
@@ -77,6 +77,13 @@ protected String getTypeName() {
                     } else {
                         return Future.succeededFuture(topic);
                     }
+                case CONTROL:
+                    if (MqttQoS.EXACTLY_ONCE.equals(message.qosLevel())) {
+                        // client tries to send control message using QoS 2
+                        return Future.failedFuture(new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST, "QoS 2 not supported for command response messages"));
+                    } else {
+                        return Future.succeededFuture(topic);
```

Comment:
do we actually support both QoS 0 and 1?

Reply:
Yes